openfl-workspace/keras_cnn_mnist/src/tfmnist_inmemory.py

```diff
@@ -26,6 +26,10 @@ def __init__(self, data_path, batch_size, **kwargs):
         #  what index/rank is this collaborator.
         # Then we have a way to automatically shard based on rank and size of
         # collaborator list.
+        try:
+            int(data_path)
+        except:
+            raise ValueError("Pass shard number using data path flag as an int.")
```

Comment:
General: `Exception`s, warnings and assertions should inform the user precisely what went wrong. In this case, one could rephrase it as 

```python
ValueError("Expected `%s` to be representable as `int`.", data_path)
```

More [examples](https://google.github.io/styleguide/pyguide.html#3102-error-messages)

Reply:
Updated in c6e35bf304c47b1ad02e73ad4f0a294bbb53721c.